openmetadata-service/src/main/java/org/openmetadata/service/resources/docstore/DocStoreResource.java

```diff
@@ -320,9 +327,40 @@ public Response createOrUpdate(
       @Context SecurityContext securityContext,
       @Valid CreateDocument cd) {
     Document doc = getDocument(cd, securityContext.getUserPrincipal().getName());
+
+    Response emailTemplateValidationResponse = null;
+    if (doc.getEntityType().equals(DefaultTemplateProvider.ENTITY_TYPE_EMAIL_TEMPLATE)) {
```

Comment:
move this condition to validateTemplate and if the entityType is not EMAIL_TEMPLATE return

Reply:
Noted.